pkg/file/buffer_test.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package file_test
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/ethersphere/bee/pkg/file"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+// TestChunkPipe verifies that the reads are correctly buffered for
+// two unaligned writes across two chunks.
+func TestChunkPipe(t *testing.T) {
+	buf := file.NewChunkPipe()
+
+	errC := make(chan error)
+	go func() {
+		data := make([]byte, swarm.ChunkSize)
+		c, err := buf.Read(data)
+		if err != nil {
+			errC <- err
+		}
+		if c != swarm.ChunkSize {
+			errC <- fmt.Errorf("short read %d", c)
+		}
+		c, err = buf.Read(data)
+		if c != 2 {
+			errC <- fmt.Errorf("read expected 2, got %d", c)
+		}
+		if err != nil {
+			errC <- err
+		}
+		errC <- nil
+	}()
+	data := [swarm.ChunkSize - 2]byte{}
```

Comment:
1. this is extremely ugly :) can you please switch to `data := make([]byte,ChunkSize-2)`/`var data [ChunkSize-2]byte`?
2. why the 2?